fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/derby/DerbyListSequencesForSchema.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.database.utils.derby;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.ibm.fhir.database.utils.api.IDatabaseSupplier;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.common.SchemaInfoObject;
+import com.ibm.fhir.database.utils.common.SchemaInfoObject.Type;
+
+/**
+ * DAO to fetch the names of sequences in the given schema
+ */
+public class DerbyListSequencesForSchema implements IDatabaseSupplier<List<SchemaInfoObject>> {
+    
+    // The schema of the table
+    private final String schemaName;
+
+    /**
+     * Public constructor
+     * @param schemaName
+     */
+    public DerbyListSequencesForSchema(String schemaName) {
+        this.schemaName = DataDefinitionUtil.assertValidName(schemaName);
+    }
+
+    @Override
+    public List<SchemaInfoObject> run(IDatabaseTranslator translator, Connection c) {
+        List<SchemaInfoObject> result = new ArrayList<>();
+        // Grab the list of sequences for the configured schema from the Derby sys catalog
+        final String sql = ""
+                + "SELECT seqs.sequencename FROM sys.syssequences AS seqs "
+                + "  JOIN sys.sysschemas AS schemas "
+                + "    ON (seqs.schemaid = schemas.schemaid) "
+                + " WHERE schemas.schemaname = ?"
+                ;
+        
+        try (PreparedStatement ps = c.prepareStatement(sql)) {
+            ps.setString(1, schemaName);
+            ResultSet rs = ps.executeQuery();
+            while (rs.next()) {
+                result.add(new SchemaInfoObject(Type.SEQUENCE, rs.getString(1)));
+            }
+        }
+        catch (SQLException x) {
+            throw translator.translate(x);
+        }
```

Comment:
Identical pattern across 9 classes. 
Perhaps AbstractListSchema with a method to add to results?